File: cmux-tui/bindings/java/src/com/cmux/TopologyResnapshotReason.java
package com.cmux;

public enum TopologyResnapshotReason {
    STALE_DAEMON("stale-daemon"), STALE_SESSION("stale-session"),
    REVISION_AHEAD("revision-ahead"), HISTORY_GAP("history-gap"),
    REPLAY_TOO_LARGE("replay-too-large"), SLOW_CONSUMER("slow-consumer");

    private final String wire;
    TopologyResnapshotReason(String wire) { this.wire = wire; }
    public String wire() { return wire; }

    static TopologyResnapshotReason from(String wire) {
        for (TopologyResnapshotReason value : values()) if (value.wire.equals(wire)) return value;
        throw new IllegalArgumentException("invalid topology resnapshot reason " + wire);
    }
}

File: Packages/macOS/CmuxTerminalBackend/Sources/CmuxTerminalBackend/TopologyResnapshotReason.swift
/// The reason a topology stream cannot safely continue from its requested revision.
public enum TopologyResnapshotReason: String, Codable, Equatable, Sendable {
    /// The client named a daemon lifetime that is no longer current.
    case staleDaemon = "stale-daemon"

    /// The client named a persisted session that is no longer current.
    case staleSession = "stale-session"

    /// The client requested a revision newer than the daemon's current revision.
    case revisionAhead = "revision-ahead"

    /// The daemon no longer retains every delta needed to resume.
    case historyGap = "history-gap"

    /// The required replay exceeds the daemon's bounded replay limit.
    case replayTooLarge = "replay-too-large"

    /// The client consumed topology events too slowly to preserve continuity.
    case slowConsumer = "slow-consumer"
}
